x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/enrollment/tool/BootstrapPasswordAndEnrollmentTokenForInitialNodeTests.java

```diff
@@ -0,0 +1,318 @@
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BootstrapPasswordAndEnrollmentTokenForInitialNodeTests extends ESTestCase {
+    private Environment environment;
+
+    @BeforeClass
+    public static void muteInFips(){
+        assumeFalse("Enrollment is not supported in FIPS 140-2 as we are using PKCS#12 keystores", inFipsJvm());
+    }
+
+    @Before
+    public void setupMocks() throws Exception {
+        final Path tempDir = createTempDir();
+        final Path httpCaPath = tempDir.resolve("httpCa.p12");
+        Files.copy(getDataPath("/org/elasticsearch/xpack/security/action/enrollment/httpCa.p12"), httpCaPath);
+        final MockSecureSettings secureSettings = new MockSecureSettings();
+        secureSettings.setString("xpack.http.ssl.truststore.secure_password", "password");
+        secureSettings.setString("xpack.security.http.ssl.keystore.secure_password", "password");
+        secureSettings.setString("keystore.seed", "password");
+        final Settings settings = Settings.builder()
+            .put("xpack.security.enabled", true)
+            .put("xpack.http.ssl.enabled", true)
+            .put("xpack.security.authc.api_key.enabled", true)
+            .put("xpack.http.ssl.truststore.path", "httpCa.p12")
+            .put("xpack.security.http.ssl.enabled", true)
+            .put("xpack.security.http.ssl.keystore.path", "httpCa.p12")
+            .put("xpack.security.enrollment.enabled", "true")
+            .setSecureSettings(secureSettings)
+            .put("path.home", tempDir)
+            .build();
+        environment = new Environment(settings, tempDir);
+    }
+
+    public void testGenerateNewPasswordSuccess() throws Exception {
+        final BootstrapPasswordAndEnrollmentTokenForInitialNode command_util = mock(BootstrapPasswordAndEnrollmentTokenForInitialNode
+            .class);
+        final CommandLineHttpClient client_util = mock(CommandLineHttpClient.class);
+        doReturn(client_util).when(command_util).getClient(environment);
+        final MockTerminal terminal = new MockTerminal();
+        final CommandLineHttpClient client = new CommandLineHttpClient(environment);
+        doReturn(client.getDefaultURL()).when(client_util).getDefaultURL();
+        final EnrollmentTokenGenerator enrollmentTokenGenerator = mock(EnrollmentTokenGenerator.class);
+        final EnrollmentToken token = new EnrollmentToken("DR6CzXkBDf8amV_48yYX:x3YqU_rqQwm-ESrkExcnOg",
+            "ce480d53728605674fcfd8ffb51000d8a33bf32de7c7f1e26b4d428f8a91362d", "8.0.0",
+            Arrays.asList("[192.168.0.1:9201, 172.16.254.1:9202"));
+        final BootstrapPasswordAndEnrollmentTokenForInitialNode command = new BootstrapPasswordAndEnrollmentTokenForInitialNode() {
```

Comment:
we do we have both a mock (command_util) and an implementation of BootstrapPasswordAndEnrollmentTokenForInitialNode ?

Reply:
Yes, I want to be able to reach the real methods implementations (like checkClusterHealth) while mocking the main command.

Reply:
I'm not sure you need to do this. For instance, you don't need to do 
```
final BootstrapPasswordAndEnrollmentTokenForInitialNode command_util = mock(BootstrapPasswordAndEnrollmentTokenForInitialNode
            .class);
doReturn(client_util).when(command_util).getClient(environment);
```
, you are already doing this in 

```
@Override
protected CommandLineHttpClient getClient(Environment env) {
    return client_util;
}
```

a few lines below. You can see that if you remove the first 2 lines, the test still passes.

Reply:
I refactored the tests, didn't need it indeed